### PR TITLE
Delete latedays model & added latedays table view

### PR DESCRIPTION
**Pull request by saxocellphone**
Now all late days related functions will use the new method.

### CODE REVIEW

**Comment by saxocellphone** [comment]
I did some testing, everything seems to work fine besides the GenerateGradereport. However, the same error occurs on the master branch, so I don't think I had anything to do with it.

**Comment by saxocellphone** [comment]
Also for the SubmissionControllerTester I deleted the latedays related lines but now it's breaking. I'm not quite sure how to approach the problem

**Comment by jchulton** [comment]
Minor errors: Lets say a student has 3 late days and goes to submit 1 day late. Prior to submission, the prompt says that they will be charged one late day and have three remaining. This is incorrect, they would have 2. If the student submits, then goes to resubmit on the same assignment the prompt says they would have 2 late days remaining. This is technically correct, but the # of late days shouldn't change for resubmitting on an assignment which the student already used a late day on.

**Comment by jchulton** [comment]
It looks like the tests are failing on line 77 on Submitty\site/app/controllers/student/SubmissionController.php
It looks like there is a case where gradeable_id is null

File: site/app/views/grading/ElectronicGraderView.php [hunk 1]
```diff
@@ -807,19 +806,23 @@ private function makeTable($user_id, $gradeable, &$status){
             <tbody>
 HTML;
         $total_late_used = 0;
+<<<<<<< HEAD
```
Comment (CouleeApps) [inline site/app/views/grading/ElectronicGraderView.php]: Nice merge conflict

**Comment by CouleeApps** [comment]
You're failing a test: https://travis-ci.org/Submitty/Submitty/jobs/390876092

**Comment by jchulton** [comment]
Looks good except for one minor banner error: if a student has no late days and goes to submit, if the assignment allows for late days, the banner will read the following, despite the student not having enough late days.
lateday
I don't think this bug should hold up the pull request, since the purpose of this pull request was to remodel how late days are handled, and late days are being calculated properly on the instructor side. I opened issue #2153 to fix the banner.

Also, is Travis now in a working state?

**Comment by saxocellphone** [comment]
Oops forgot to take into account those people who are late instead of good.

**Comment by bmcutler** [comment]
@saxocellphone please resolve these conflicts

**Comment by CouleeApps** [comment]
Considering it's a view I changed in an earlier PR, I should resolve these :)

File: site/app/controllers/student/LateDaysTableController.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace app\controllers\student;
+
+
+use app\controllers\AbstractController;
+use app\libraries\Core;
+
+class LateDaysTableController extends AbstractController {
+    public function run() {
+        $g_id = isset($_REQUEST["g_id"]) ? $_REQUEST["g_id"] : NULL;
```
Comment (MasterOdin) [inline site/app/controllers/student/LateDaysTableController.php]: $g_id = $_REQUEST['g_id'] ?? null;

**Pull request merged**